### PR TITLE
chore: update oss/ee oidc & saml helm config

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
In the user auth project, we only added the helm edits to EE -- this PR ports them back to OSS to standardize the oidc/saml helm config.

tldr; haran noticed that the helm changes were in EE only, but should be in OSS, plus all the user auth provisioning features hadn't been properly included there.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: helm/charts/determined/values.yaml [hunk 1]
```diff
@@ -125,6 +128,20 @@ useNodePortForMaster: false
...
+# provider.
+# saml:
+#   enabled:
+#   provider:
+#   idpRecipientUrl:
+#   idpSsoUrl:
+#   idpSsoDescriptorUrl:
+#   idpCertPath:
+#   autoProvisionUsers:
+#   groupsAttributeName:
```
Comment (salonig23) [inline helm/charts/determined/values.yaml:142]: should be `groupsClaimName` to make it consistent

File: helm/charts/determined/templates/master-config.yaml [hunk 2]
```diff
@@ -106,6 +115,25 @@ stringData:
...
+      idp_sso_descriptor_url: {{ required "A valid sso descriptor url is required!" .Values.saml.idpSsoDescriptorUrl }}
+      idp_cert_path: {{ required "A valid idp cert path is required!" .Values.saml.idpCertPath }}
+      {{- if .Values.saml.autoProvisionUsers }}
+      auto_provision_users: {{ .Values.saml.autoProvisionUsers }}
+      {{- end }}
+      {{- if .Values.saml.groupsAttributeName }}
+      groups_claim_name: {{ .Values.saml.groupsAttributeName }}
+      {{- end }}
+      {{- if .Values.saml.displayNameAttributeName }}
+      display_name_attribute_name: {{ .Values.saml.displayNameAttributeName }}
```
Comment (salonig23) [inline helm/charts/determined/templates/master-config.yaml:134]: Maybe it is better to make it `display_name_claim_name` if we are changing `groups_attribute_name` to `groups_claim_name`

Same change in values.yaml too

**Review by salonig23 — CHANGES_REQUESTED** [review]

File: helm/charts/determined/values.yaml (hunk 1 above)
Comment (carolinaecalderon) [reply helm/charts/determined/values.yaml:142]: i can put up a separate patch PR for that, as changing the variable names would probably require changing the docs & other files as well

**Review by salonig23 — APPROVED** [review]
LGTM

**Pull request merged**